libavfilter/dnn/dnn_backend_tf.c

```diff
@@ -242,6 +239,12 @@ static DNNReturnType load_tf_model(TFModel *tf_model, const char *model_filename
     tf_model->graph = TF_NewGraph();
     tf_model->status = TF_NewStatus();
     graph_opts = TF_NewImportGraphDefOptions();
+    if(deviceid >= 0x30) {
+        sprintf(sdevice,"/gpu:%d", (int)(deviceid-0x30));
+        TF_ImportGraphDefOptionsSetDefaultDevice(graph_opts, sdevice);
+        //restore default value
+        deviceid = -1;
```

Comment:
Looks like the caller has to call `ff_dnn_set_deviceid_tf` before loading a model otherwise `deviceid = -1` and we won't set a default device here. Is that right? Or is there different default behavior when the caller doesn't call `ff_dnn_set_deviceid_tf`?

Reply:
Right, if deviceid = -1 then tensorflow select device 0 by default. so all model loaded on device 0.